Assets/Scripts/Core/GameManager.cs

```diff
@@ -0,0 +1,180 @@
+﻿using Assets.Scripts.Model;
+using UnityEngine;
+using UnityStandardAssets.Characters.FirstPerson;
+
+public class GameManager : MonoBehaviour {
+
+    public Configurator configurator;
+
+    public Player playerPrefab;
+
+    public static GameManager instance = null;
+
+    private GameState state;
+
+    private int levelNumber;
+
+    private Player player;
+
+    // Awake is always called before any Start functions
+    void Awake()
+    {
+        if (instance == null)
+        {
+            instance = this;
+        }
+        else if (instance != this)
+        {
+            Destroy(gameObject);
+        }
+        DontDestroyOnLoad(gameObject);  // Sets this to not be destroyed when reloading scene
+
+        Init();
+
+        SwitchGameState(configurator.initialApplicationState);
+    }
+
+    private void Init()
+    {
+        levelNumber = configurator.initialLevelNumber;
+        player = Instantiate(playerPrefab) as Player;
+        player.name = "Player";
+        player.remainingLives = configurator.initialPlayerLives;
+        player.bombs = configurator.initialPlayerBombs;
+        player.bombRange = configurator.initialPlayerBombRange;
+        player.speed = configurator.initialPlayerSpeed;
+        player.remoteDetonationBonus = configurator.initialPlayerRemoteDetonationBonus;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+
+    }
+
+    /****************************************/
+    /*          Public API Methods          */
+    /****************************************/
+
+    public void SwitchGameState(GameState state)
+    {
+        this.state = state;
```

Comment:
Potrzebujemy przechowywać stan? Będzie on gdzieś potem potrzebny? Bo jak nie to może ta metoda działać na swoim argumencie.